### PR TITLE
Select cachetool version based on PHP version

### PR DESCRIPTION
**Pull request by aegypius**
- [x] Stores PHP version in a fact to be reused for later
- [x] Guess Cachetool version against PHP version 
- [x] Guess Cachetool options against PHP version 
- [x] If Cachetool version is lower than the required playbook update Cachetool to the required version

Superseed #15

### CODE REVIEW

File: config/steps/cachetool.yml [hunk 1]
```diff
@@ -1,21 +1,42 @@
...
 - name: LEPHARE - Check for cachetool.phar
   stat: path={{ lephare_cachetool_path }}
   register: cachetool_file
 
 - name: LEPHARE - Run cachetool self-update
+  ignore_errors: true
   shell: "{{ lephare_cachetool_path }} self-update"
-  when: cachetool_file.stat.exists and lephare_cachetool_self_update
-  ignore_errors: True
+  when: cachetool_file.stat.exists and lephare_cachetool_self_update and cachetool_facts.version == 'latest'
```
Comment (pierreboissinot) [inline config/steps/cachetool.yml:25]: why `cachetool_facts.version == 'latest'` ? 
a cachetool 5.1.3 should upgrade to 5.1.4 if available ?

File: config/steps/cachetool.yml [hunk 2]
```diff
@@ -1,21 +1,42 @@
...
   register: cachetool_self_update_result
   changed_when: cachetool_self_update_result.stderr
 
-- name: LEPHARE - Install cachetool
+- name: LEPHARE - Install latest cachetool
   get_url: url=http://gordalina.github.io/cachetool/downloads/cachetool.phar dest={{ lephare_cachetool_path }} mode=0755 validate_certs=no force=no
-  when: not cachetool_file.stat.exists
+  when: not cachetool_file.stat.exists and cachetool_facts.version == 'latest'
```
Comment (pierreboissinot) [inline config/steps/cachetool.yml:31]: latest case could be included in the bask below no ?

**Review by pierreboissinot — APPROVED** [review]

File: config/steps/cachetool.yml (hunk 1 above)
Comment (aegypius) [reply config/steps/cachetool.yml:41]: @pierreboissinot for the record, cachetool does not check php requirements when self-update is called. This is why self-update task is called only if the `latest` version is expected

**Comment by aegypius** [comment]
@pierreboissinot Here is an updated version that should handle nicely the upgrade to the latest required version if cachetool is outdated.

**Review by thislg — APPROVED** [review]

**Review by pierreboissinot — APPROVED** [review]

**Pull request merged**